src/main/scala/tech/cryptonomic/conseil/tezos/JsonDecoders.scala

```diff
@@ -224,6 +227,26 @@ object JsonDecoders {
       implicit val transactionMetadataDecoder: Decoder[ResultMetadata[OperationResult.Transaction]] = deriveDecoder
       implicit val originationMetadataDecoder: Decoder[ResultMetadata[OperationResult.Origination]] = deriveDecoder
       implicit val delegationMetadataDecoder: Decoder[ResultMetadata[OperationResult.Delegation]] = deriveDecoder
+      implicit val internalOpResultsDecoder: Decoder[InternalOperationResults.InternalOperationResult] =
```

Comment:
Just wondering if it might work

Considering that
- there's a custom derivation config using "kind" as discriminator field,
- we have a separate object/module scope than the "external" operations, to avoid name clashes

Do you think that renaming the internal operation results exactly as their kind could allow for a single derivation, i.e.

```scala
implicit val internalOperationResultDecoder: Decoder[InternalOperationResult.InternalOperationResult] =
  deriveDecoder
```
Letting circe handle the ADT hierarchy?

Reply:
unfortunately I couldn't make this circe derivation work, thats why I used this custom decoder

Reply:
nvm, apparently I've done something wrong before, now it works, thanks!